x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key.ts

```diff
@@ -4,171 +4,77 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import uuid from 'uuid';
-import Boom from '@hapi/boom';
-import { SavedObjectsClientContract, SavedObject } from 'src/core/server';
-import { EnrollmentAPIKey, EnrollmentAPIKeySOAttributes } from '../../types';
-import { ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE } from '../../constants';
-import { createAPIKey, invalidateAPIKey } from './security';
-import { agentPolicyService } from '../agent_policy';
+import { SavedObjectsClientContract, ElasticsearchClient } from 'src/core/server';
+import { EnrollmentAPIKey } from '../../types';
 import { appContextService } from '../app_context';
-import { normalizeKuery } from '../saved_object';
+import * as enrollmentApiKeyServiceSO from './enrollment_api_key_so';
+import * as enrollmentApiKeyServiceFleetServer from './enrollment_api_key_fleet_server';
 
 export async function listEnrollmentApiKeys(
   soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
   options: {
     page?: number;
     perPage?: number;
     kuery?: string;
     showInactive?: boolean;
   }
 ): Promise<{ items: EnrollmentAPIKey[]; total: any; page: any; perPage: any }> {
-  const { page = 1, perPage = 20, kuery } = options;
-
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  const { saved_objects, total } = await soClient.find<EnrollmentAPIKeySOAttributes>({
-    type: ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE,
-    page,
-    perPage,
-    sortField: 'created_at',
-    sortOrder: 'desc',
-    filter:
-      kuery && kuery !== ''
-        ? normalizeKuery(ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE, kuery)
-        : undefined,
-  });
-
-  const items = saved_objects.map(savedObjectToEnrollmentApiKey);
-
-  return {
-    items,
-    total,
-    page,
-    perPage,
-  };
+  if (appContextService.getConfig()?.agents?.fleetServerEnabled === true) {
+    return enrollmentApiKeyServiceFleetServer.listEnrollmentApiKeys(esClient, options);
+  } else {
+    return enrollmentApiKeyServiceSO.listEnrollmentApiKeys(soClient, options);
+  }
```

Comment:
nit: we could DRY up these conditionals by writing another method like
```ts
function getEnrollmentApiKeyService() {
  if (appContextService.getConfig()?.agents?.fleetServerEnabled === true) {
    return enrollmentApiKeyServiceFleetServer;
  } else {
    return enrollmentApiKeyServiceSO;
  }
}
```

and use like `getEnrollmentApiKeyService().listEnrollmentApiKeys(...)`

Reply:
I know it's a little more verbose but it's temporary as it's in the end the `*_crud` service will disappear and this way we do not change how the service is consumed